doc/source/development/rfc/rfc102_embedded_resources.rst

```diff
@@ -0,0 +1,170 @@
+===================================================================
+
+============== =============================================
+Author:        Even Rouault
+Contact:       even.rouault @ spatialys.com
+Started:       2024-Oct-01
+Status:        Draft
+Target:        GDAL 3.10 or 3.11
+============== =============================================
+
+Summary
+-------
+
+This RFC uses C23 ``#embed`` pre-processor directive, when available,
+to be able to embed GDAL resource files directly into libgdal. It is also
+intended to be used for PROJ, in particular for its :file:`proj.db` file.
+
+Motivation
+----------
+
+Some parts of GDAL core, but mostly drivers, depend on a number of resource
+files for correct execution. Locating those resource files on the filesystem
+can be painful in some use cases of GDAL, that involve relocating the GDAL
+binary at installation time. One such case could be the GDAL embedded in Rasterio
+or Fiona binary wheels where :config:`GDAL_DATA` must be correctly set currently.
+Web-assembly (WASM) use cases come also to mind as users of GDAL builds where
+resources are directly included in libgdal.
+
+Technical solution
+------------------
+
+The C23 standard includes a `#embed "filename" <https://en.cppreference.com/w/c/preprocessor/embed>`__
+pre-processor directive that ingests the specified filename and returns its
+content as tokens that can be stored in a unsigned char or char array.
+
+Getting the content of a file into a variable is as simple as the following
+(which also demonstrates adding a nul-terminating character when this is needed):
+
+.. code-block:: c
+
+    static const char szPDS4Template[] = {
+    #embed "data/pds4_template.xml"
+        , '\0'};
+
+Compiler support
+----------------
+
+Support for that directive is still very new. clang 19.1 is the
+first compiler which has a release including it, and has an efficient
+implementation of it, able to embed very large files with minimum RAM and CPU
+usage.
+
+The development version of GCC 15 also supports it, but in a non-optimized way
+for now. i.e. trying to include large files, of several tens of megabytes could
+cause significant compilation time, but without impact on runtime. This is not
+an issue for GDAL use cases, and there is intent from GCC developers to improve
+this in the future.
+
+Embedding PROJ's :file:`proj.db` of size 9.1 MB with GCC 15dev at time of writing takes
+18 seconds and 1.7 GB RAM, compared to 0.4 second and 400 MB RAM for clang 19,
+which is still reasonable (Generating :file:`proj.db` itself from its source .sql files
+takes one minute on the same system).
+
+There is no timeline for Visual Studio C/C++ at time of writing (it has been
+`requested by users <https://developercommunity.visualstudio.com/t/Add-support-for-embed-as-voted-into-the/10451640?ru=https%3A%2F%2Fdevelopercommunity.visualstudio.com%2Fcontent%2Fproblem%2F132967%2Fcopycut-commands-doesnt-work.html%3FchildToView%3D133119>`__)
```

Comment:
Can the URL be shortened to https://developercommunity.visualstudio.com/t/Add-support-for-embed-as-voted-into-the/10451640 ?

Reply:
done, thanks